### ISSUE TITLE
Redis client filters the null values for with the mget command

### ISSUE THREAD

**Issue by Plawn** [quarkusio/quarkus#28049]
### Describe the bug

In the redis documentation, here : https://redis.io/commands/mget/

The missing values should be returned as null, it enables to detect in an efficient way, which keys are missing or have been deleted due to timeout policies.

However, the implementation of the ReactiveRedisClient uses the io.quarkus.redis.runtime.datasource.Marshaller which filters this values with the method #decodeAsOrderedMap. 

The null values should not be filtered and the user should do it, if necessary.

### Expected behavior

More compliance with the redis documentation, and non filtered null values.

### Actual behavior

The null values are filtered by default.

### How to Reproduce?

1. Create a quarkus app with version 2.12.2.Final
2. Create a ReactiveRedisDatasource
3. use the mget command to fetch non existing values
4. Recover an empty map

### Quarkus version or git rev

2.12.2.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

maven

**Comment by cescoffier** [comment]
One solution would be to have the entries in the map (key -> `null`). Would that work for you?

**Comment by Plawn** [comment]
Yes thats perfect !, we just have to remove the null check, it should not be breaking anything as the filter was not stated in the documentation
